components/hubspot/sources/new-contact-in-list/new-contact-in-list.js

```diff
@@ -25,28 +36,34 @@ module.exports = {
     async emitEvent(contact, properties, list) {
       const contactInfo = await this.hubspot.getContact(
         contact.vid,
-        properties
+        properties,
       );
       const meta = this.generateMeta(contact, list);
-      this.$emit({ contact, contactInfo }, meta);
+      this.$emit({
+        contact,
+        contactInfo,
+      }, meta);
     },
-  },
-  async run(event) {
-    const properties = this.db.get("properties");
-    for (let list of this.lists) {
-      list = JSON.parse(list);
-      const params = {
+    getParams() {
+      return {
         count: 100,
       };
-      let hasMore = true;
-      while (hasMore) {
-        const results = await this.hubspot.getListContacts(params, list.value);
-        hasMore = results["has-more"];
-        if (hasMore) params.vidOffset = results["vid-offset"];
-        for (const contact of results.contacts) {
-          await this.emitEvent(contact, properties, list);
+    },
+    async processResults() {
+      const properties = this._getProperties();
+      for (let list of this.lists) {
+        list = JSON.parse(list);
+        const params = this.getParams();
+        let hasMore = true;
+        while (hasMore) {
```

Comment:
The endpoint for this one is from v1, so the pagination is a bit different. It uses results["has-more"] instead of results.hasMore, results["vid-offset"] instead of results.offset, and this module passes multiple parameters into emitEvent so that we have the list name to use in generateMeta.